Design.md

```diff
@@ -0,0 +1,241 @@
+   + [Query Tenants](#query-tenants)
+   + [List Tenants](#list-tenants)
+   + [Get Tenant](#get-tenant)
+   + [Head Tenant](#head-tenant)
+   + [Delete Tenant](#delete-tenant)
+   + [Update Tenant](#update-tenant)
+* [User APIs](#user-apis)
+  + [Create User](#create-user)
+  + [Create User Example Activity Diagram](#create-user-example-activity-diagram)
+  + [Query Users](#query-users)
+  + [List Users](#list-users)
+  + [Get User w/ userid](#get-user-w--userid)
+  + [Get User w/ canonical ID](#get-user-w--canonical-id)
+  + [Head User](#head-user)
+  + [Delete User](#delete-user)
+  + [Update User](#update-user)
+* [S3 Credential APIs](#s3-credential-apis)
+  + [Create S3 Credential](#create-s3-credential)
+  + [Query S3 Credentials](#query-s3-credentials)
+  + [List S3 Credentials](#list-s3-credentials)
+  + [Delete S3 Credential](#delete-s3-credential)
+  + [Get S3 Credential](#get-s3-credential)
+* [Miscellaneous APIs:](#miscellaneous-apis-)
+  + [getProviderConsoleUrl](#getproviderconsoleurl)
+  + [getTenantConsoleUrl](#gettenantconsoleurl)
+  + [getS3Capabilities](#gets3capabilities)
+  + [getInformation](#getinformation)
+  + [updateOsisCaps (?)](#updateosiscaps----)
+  + [getBucketList](#getbucketlist)
+  + [getOsisUsage](#getosisusage)
+* [S3 Operations Example Activity Diagram](#s3-operations-example-activity-diagram)
+</details>
+
+## OSIS inital setup
+
+For the initial installation of OSIS application, following steps must be executed.
+1. A **Super Admin** account will be created as part of S3C/Vault setup
+1. Set OSE properties:
+   1. S3 URL
+   1. Admin Vault URL along with credentials as Super Admin access key/ secret key
+    
+_**Note: "Super Admin" account has full Account Management access and can use "AssumeRoleBackbeat" to assume a role of another Account.**_
+
+## Tenant APIs
+<u>**Important Assumptions:**</u>
+* All the Tenant APIs on OSIS will use **Super Admin credentials** to manage on Vault.
+* A Vault Account is equivalent of OSE Tenant.
+
+### Create Tenant
+This API creates a Tenant on Vault.  
+1. Vault `create-account` api will be called using vaultclient. 
+    * Tenant Name is stored as Account Name  
+    * `cd_tenant_ids` list as part of the request to be stored as a property of the vault account.
+    * Account ID to be sent back as storage tenant ID to OSE
+1. `generate-account-access-key` for the account using vaultclient.
+1. Create a role with
+   1. Role name : `osis` (So role-arn can be generated in the format `arn:aws:iam::[account-id]:role/osis`
+   1. Trust policy
+   ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [{
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "service-name"
+            },
+            "Action": "sts:AssumeRole"
+            }]
+    }
+   ```
+1. `attach-role-policy` with s3 and iam full access to this role
+1. `delete-access-key` for the account.
```

Comment:
Maybe you already know. But `DeleteAccessKey`, is an IAM call. So the AK/SK that was generated for the account will be used for deleting the access key as well.

Hence, this is one call, which will be made using account credentials.
Vault SuperAdmin credentials cannot make this call.

Reply:
Sure we can do that too but will it not be inconsistent if we do other apis with assumed account and only this api with super admin?

Reply:
Updated in commit [59a5a88](https://github.com/scality/vmware-ose-scality/pull/11/commits/59a5a884aa427cfb91917ac2bc196ee07651ebd8)